rotkehlchen/history/events/structures/solana_event.py

```diff
@@ -0,0 +1,199 @@
+import logging
+from typing import TYPE_CHECKING, Any, cast
+
+from solders.solders import Signature
+
+from rotkehlchen.accounting.types import EventAccountingRuleStatus
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.history.events.structures.base import (
+    HISTORY_EVENT_DB_TUPLE_WRITE,
+    HistoryBaseEntry,
+    HistoryBaseEntryType,
+    HistoryEventWithCounterparty,
+)
+from rotkehlchen.history.events.structures.types import (
+    CHAIN_EVENT_FIELDS_TYPE,
+    HistoryEventSubType,
+    HistoryEventType,
+)
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_fval, deserialize_optional
+from rotkehlchen.types import (
+    FVal,
+    Location,
+    SolanaAddress,
+    TimestampMS,
+)
+from rotkehlchen.utils.misc import timestamp_to_date, ts_ms_to_sec
+
+if TYPE_CHECKING:
+
+    from rotkehlchen.history.events.structures.types import SOLANA_EVENT_DB_TUPLE_READ
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class SolanaEvent(HistoryEventWithCounterparty):  # hash in superclass
+    """Represents a solana event, extending HistoryBaseEntry with solana specific metadata.
+
+    Additional fields:
+    - counterparty: optional protocol identifier (e.g., 'raydium', 'orca').
+    - address: optional program address for filtering events by specific
+    programs or protocol versions.
```

Comment:
just to indent it since it belongs to the bullet point

```suggestion
        programs or protocol versions.
```